src/utils/efd_win.inc

```diff
@@ -201,14 +202,35 @@ wsafail3:
     return -rc;
 }
 
+void nn_efd_stop (struct nn_efd *self)
+{
+    int rc;
+    SOCKET s = self->w;
+    self->w = INVALID_SOCKET;
```

Comment:
Atomic swap would be a helpful wrapper utility. We have atomic increment/decrement utility functions: https://github.com/nanomsg/nanomsg/blob/master/src/utils/atomic.c